File: src/Homework1/Homework1/Calculator.cs
using System;

namespace Homework1
{
    public enum SupportedOperations
    {
        Plus,
        Minus,
        Multiply,
        Divide
    }

    public static class Calculator
    {
        public static double Calculate(SupportedOperations operation, int val1, int val2)
        {
            var result = 0.0;
            switch (operation)
            {
                case SupportedOperations.Plus:
                    result = val1 + val2;
                    break;
                case SupportedOperations.Minus:
                    result = val1 - val2;
                    break;
                case SupportedOperations.Multiply:
                    result = val1 * val2;
                    break;
                case SupportedOperations.Divide:
                    if (val2 != 0)
                        result = (double)val1 / val2;
                    break;
            }

            return result;
        }
    }
}

File: src/Homework5/Homework5/Calculator.fs
module Homework5.Calculator

open Homework5
open MaybeBuilder

type SupportedOperations =
    | Plus = 1
    | Minus = 2
    | Divide = 3
    | Multiply = 4
    
let calculate (val1 : decimal, operation, val2) =
    (*let division val1 val2 =
        if val1 = decimal 0 then
            Error "Division by zero"
        else
            Ok(val2 / val1)*)
            
    result {
        match operation with
        | SupportedOperations.Plus -> return val1 + val2
        | SupportedOperations.Minus -> return val1 - val2
        | SupportedOperations.Multiply -> return val1 * val2
        | SupportedOperations.Divide -> return val1/ val2
    }